### DISCUSSION TITLE
关于操作步骤采用有序列表的想法

### DISCUSSION THREAD

**Discussion by airinghost** [Anduin2017/HowToCook#1091] [Ideas]
以西红柿炒鸡蛋为例，由于菜谱中的操作步骤采用的是无序列表，因此这里提到的`第10步`得读者一个一个数下来，感觉不太好。

https://github.com/Anduin2017/HowToCook/blob/fc59d88883724d3c8f9b750e61dd346a51d6f83d/dishes/vegetable_dish/%E8%A5%BF%E7%BA%A2%E6%9F%BF%E7%82%92%E9%B8%A1%E8%9B%8B.md?plain=1#L50-L52

下面是原例，可见直观上无法立刻找到`第10步`：

- 西红柿洗净
- 可选：去掉西红柿的外表皮
  - 开水烫表皮，然后将西红柿放入冷水，剥去外皮
- 西红柿去蒂，切成边长不超过 4cm 的小块，即为 `西红柿块`
- 将鸡蛋打入碗中，加入 `1g * 份数` 的盐，搅匀，即为 `鸡蛋液`
  - 可以考虑向鸡蛋中加入 1ml 醋，这可以去除腥味，令鸡蛋更蓬松
- 热锅，加入食用油
- 油热后，倒入 `鸡蛋液`。翻炒至鸡蛋结为固体且颜色微微发黄，即为 `半熟鸡蛋`
- 关火。将 `半熟鸡蛋` 盛盘，重新开火
  - 注意：不要洗锅
- 加入 `西红柿块`，锅铲拍打并翻炒 20 秒，或至西红柿软烂
- 向锅中加入 `半熟鸡蛋`，翻炒均匀
  - 可以考虑加入 10ml 番茄酱和 50ml 清水，增加汤汁
  - 可以额外加入一些其它熟肉和材料
- 加入剩余的盐、糖（可选，如果倾向于甜味版本）、葱花（可选），翻炒均匀
- 关火，盛盘

采用有序列表的话，也方便提 issue 时指出哪一（几）步可以改进。

如：

1. 西红柿洗净
2. 可选：去掉西红柿的外表皮
   - 开水烫表皮，然后将西红柿放入冷水，剥去外皮
3. 西红柿去蒂，切成边长不超过 4cm 的小块，即为 `西红柿块`
4. 将鸡蛋打入碗中，加入 `1g * 份数` 的盐，搅匀，即为 `鸡蛋液`
   - 可以考虑向鸡蛋中加入 1ml 醋，这可以去除腥味，令鸡蛋更蓬松
5. 热锅，加入食用油
6. 油热后，倒入 `鸡蛋液`。翻炒至鸡蛋结为固体且颜色微微发黄，即为 `半熟鸡蛋`
7. 关火。将 `半熟鸡蛋` 盛盘，重新开火
   - 注意：不要洗锅
8. 加入 `西红柿块`，锅铲拍打并翻炒 20 秒，或至西红柿软烂
9. 向锅中加入 `半熟鸡蛋`，翻炒均匀
   - 可以考虑加入 10ml 番茄酱和 50ml 清水，增加汤汁
   - 可以额外加入一些其它熟肉和材料
10. 加入剩余的盐、糖（可选，如果倾向于甜味版本）、葱花（可选），翻炒均匀
11. 关火，盛盘

**Comment by Tylannn** [comment]
这个提议非常不错，支持

**Comment by it985** [comment]
复议

**Comment by Anduin2017** [comment]
要是 markdown 能够原生将列表渲染成有数字标号的列表就好了。

我刚试了试，发现：

```markdown
1. 第一项
1. 第二项
1. 第三项
```

竟然会被渲染成：

1. 第一项
1. 第二项
1. 第三项

这不完美！

是不是以后我们的菜谱步骤都要用 `1.` 来表示了

**Comment by Anduin2017** [comment]
好主意。既然上面的渲染可以得到有序列表，我在考虑一种全新的语法：

```markdown
1. 西红柿洗净
1. 可选：去掉西红柿的外表皮
    * 开水烫表皮，然后将西红柿放入冷水，剥去外皮
1. 西红柿去蒂，切成边长不超过 4cm 的小块，即为 `西红柿块`
1. 将鸡蛋打入碗中，加入 `1g * 份数` 的盐，搅匀，即为 `鸡蛋液`
    * 可以考虑向鸡蛋中加入 1ml 醋，这可以去除腥味，令鸡蛋更蓬松
1. 热锅，加入食用油
1. 油热后，倒入 `鸡蛋液`。翻炒至鸡蛋结为固体且颜色微微发黄，即为 `半熟鸡蛋`
1. 关火。将 `半熟鸡蛋` 盛盘，重新开火
    * 注意：不要洗锅
1. 加入 `西红柿块`，锅铲拍打并翻炒 20 秒，或至西红柿软烂
1. 向锅中加入 `半熟鸡蛋`，翻炒均匀
    * 可以考虑加入 10ml 番茄酱和 50ml 清水，增加汤汁
      * 注意不要加醋
    * 可以额外加入一些其它熟肉和材料
1. 加入剩余的盐、糖（可选，如果倾向于甜味版本）、葱花（可选），翻炒均匀
1. 关火，盛盘
```

可以渲染出：

1. 西红柿洗净
1. 可选：去掉西红柿的外表皮
    * 开水烫表皮，然后将西红柿放入冷水，剥去外皮
1. 西红柿去蒂，切成边长不超过 4cm 的小块，即为 `西红柿块`
1. 将鸡蛋打入碗中，加入 `1g * 份数` 的盐，搅匀，即为 `鸡蛋液`
    * 可以考虑向鸡蛋中加入 1ml 醋，这可以去除腥味，令鸡蛋更蓬松
1. 热锅，加入食用油
1. 油热后，倒入 `鸡蛋液`。翻炒至鸡蛋结为固体且颜色微微发黄，即为 `半熟鸡蛋`
1. 关火。将 `半熟鸡蛋` 盛盘，重新开火
    * 注意：不要洗锅
1. 加入 `西红柿块`，锅铲拍打并翻炒 20 秒，或至西红柿软烂
1. 向锅中加入 `半熟鸡蛋`，翻炒均匀
    * 可以考虑加入 10ml 番茄酱和 50ml 清水，增加汤汁
      * 注意不要加醋
    * 可以额外加入一些其它熟肉和材料
1. 加入剩余的盐、糖（可选，如果倾向于甜味版本）、葱花（可选），翻炒均匀
1. 关火，盛盘

**Reply by Anduin2017** [reply]
我建议提一个issue来专门追踪这件事。它非常有价值而且能够解决问题。我这两天有空了去把所有菜谱重新格式化一下。
